scaling_performance/managing_hugepages.adoc

```diff
@@ -0,0 +1,126 @@
+not recommend to use them for production. These features provide early access to
+upcoming product features, enabling customers to test functionality and provide
+feedback during the development process.
+
+For more information on Red Hat Technology Preview features support scope, see
+https://access.redhat.com/support/offerings/techpreview/.
+endif::[]
+====
+
+Memory is managed in blocks known as pages. On most systems, a page is 4Ki. 1Mi
+of memory is equal to 256 pages; 1Gi of memory is 256,000 pages, and so on. CPUs
+have a built-in memory management unit that manages a list of these pages in
+hardware. The Translation Lookaside Buffer (TLB) is a small hardware cache of
+virtual-to-physical page mappings. If the virtual address passed in a hardware
+instruction can be found in the TLB, the mapping can be determined quickly. If
+not, a TLB miss occurs, and the system falls back to slower, software-based
+address translation, resulting in performance issues. Since the size of the
+TLB is fixed, the only way to reduce the chance of a TLB miss is to increase the
+page size.
+
+A huge page is a memory page that is larger than 4Ki. On x86_64 architectures,
+there are two common huge page sizes: 2Mi and 1Gi. Sizes vary on other
+architectures. In order to use huge pages, code must be written so that
+applications are aware of them. Transparent Huge Pages (THP) attempt to automate
+the management of huge pages without application knowledge, but they have
+limitations. In particular, they are limited to 2Mi page sizes. THP can lead to
+performance degradation on nodes with high memory utilization or fragmentation
+due to defragmenting efforts of THP, which can lock memory pages. For this
+reason, some applications may be designed to (or recommend) usage of
+pre-allocated huge pages instead of THP.
+
+In {product-title}, applications in a pod can allocate and consume pre-allocated
+huge pages. This topic describes how.
+
+[[huge-pages-prerequisites]]
+== Prerequisites
+
+. Nodes must pre-allocate huge pages in order for the node to report its huge
+ page capacity. A node can only pre-allocate huge pages for a single size.
+
+. `HugePages` must be set to `true` across the system:
++
+----
+# cat /etc/origin/node/node-config.yaml
+...
+kubeletArguments:
+...
+  feature-gates:
+  - HugePages=true
+
+# systemctl restart atomic-openshift-node
+----
+
+[[consuming-huge-pages]]
+== Consuming Huge Pages
+
+Huge pages can be consumed via container level resource requirements using the
+resource name `hugepages-<size>`, where size is the most compact binary
+notation using integer values supported on a particular node. For example, if a
+node supports 2048KiB page sizes, it will expose a schedulable resource
+`hugepages-2Mi`. Unlike CPU or memory, huge pages do not support overcommitment.
+
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  generateName: hugepages-volume-
+spec:
+  containers:
```

Comment:
add the following:

```
containers:
 securityContext:
  privileged: true
```
I suspect on the majority of machines in the world will report `cat /proc/sys/vm/hugetlb_shm_group` as `0`, so you effectively need to run privileged right now.